ilc/common/wrapWithCache.js

```diff
@@ -1,63 +1,55 @@
+const wrapWithCache = (cacheStorage, logger, createHash) => (fn, cacheParams = {}) => {
     const {
-        cacheForSeconds,
+        cacheForSeconds = 60,
     } = cacheParams;
 
-    const cache = {};
     const cacheResolutionPromise = {};
 
     return (...args) => {
         const now = Math.floor(Date.now() / 1000);
-        const hash = args.length > 0 ? hashFn(JSON.stringify(args)) : '__null__';
+        const hash = args.length > 0 ? createHash(JSON.stringify(args)) : '__null__';
 
-        if (cache[hash] === undefined || cache[hash].cachedAt < now - cacheForSeconds) {
+        if (!cacheStorage.has(hash) || cacheStorage.get(hash).cachedAt < now - cacheForSeconds) {
             if (cacheResolutionPromise[hash] !== undefined) {
-                if (cache[hash] !== undefined) { // It's better to return stale cache rather then cause delay
-                    return Promise.resolve(cache[hash]);
+                if (cacheStorage.has(hash)) { // It's better to return stale cache rather then cause delay
+                    return Promise.resolve(cacheStorage.get(hash));
                 }
 
                 return cacheResolutionPromise[hash];
             }
 
             cacheResolutionPromise[hash] = fn(...args).then(data => {
                 delete cacheResolutionPromise[hash];
-                cache[hash] = {
+                cacheStorage.set(hash, {
                     data,
                     checkAfter: now + cacheForSeconds,
                     cachedAt: now,
-                };
+                });
 
-                return cache[hash];
+                return cacheStorage.get(hash);
             }).catch(err => {
                 delete cacheResolutionPromise[hash];
 
-                if (cache[hash] === undefined) {
+                if (!cacheStorage.has(hash)) {
                     return Promise.reject(err); //As someone is waiting for promise - just reject it
                 } else {
                     // Here no one waiting for this promise anymore, thrown error would cause
                     // unhandled promise rejection
                     //TODO: add better error reporting
-                    console.error('Error during cache update function execution');
-                    console.error(err);
+                    logger.error('Error during cache update function execution');
+                    logger.error(err);
                 }
             });
 
-            if (cache[hash] !== undefined) { // It's better to return stale cache rather then cause delay
-                return Promise.resolve(cache[hash]);
+            if (cacheStorage.has(hash)) { // It's better to return stale cache rather then cause delay
+                return Promise.resolve(cacheStorage.get(hash));
             }
 
             return cacheResolutionPromise[hash];
         }
 
-        return Promise.resolve(cache[hash]);
+        return Promise.resolve(cacheStorage.get(hash));
     };
 };
 
-function hashFn(s) {
```

Comment:
I would prefer to keep it here. No one should use it outside of the cache provider as it's too weak.

Reply:
Removed changes